pubber/src/main/java/daq/pubber/MqttPublisher.java

```diff
@@ -163,6 +163,9 @@ public void publish(String deviceId, String topicSuffix, Object data, Runnable c
     Object marked =
         topicSuffix.startsWith(EVENT_MARK_PREFIX) ? decorateMessage(topicSuffix, data) : data;
     try {
+      if (!isActive()) {
+        publisherExecutor = Executors.newFixedThreadPool(PUBLISH_THREAD_COUNT);
```

Comment:
Overall, I think it would just be cleaner to just not shutdown the executor in the close() method. The core of the problem seems to be that close() gets called from non-shutdown situations (like connection lost), so I'd rather keep the changes isolated to that part of the code rather than having the infection spread...

Reply:
Shutdown and close logic separated. Seems to be periodic senders failing to publish messages when connection lost but it's not handled by Pubber or corresponding manger. I think we also need to consider triggering reconnect attempt at some point (errorCounter maybe?) so that we can use backoff logic when trying to reconnect.

Currently all proxy devices and gateway trying to publish periodic events and ignoring network problems. This creates error flood on each publish attempt.